sdk/core/src/migration/credentials/index.ts

```diff
@@ -0,0 +1,208 @@
+    const publicKeyBuffer = this.keysService.getOwnPublicKey()
+    const privateKeyBuffer = this.keysService.getOwnPrivateKey()
+    const encryptedCredentials: vcMigrationList[] = []
+
+    for (const credential of credentials) {
+      let credentialId = credential?.id
+      if (!isW3cCredential(credential)) {
+        credentialId = credential?.data?.id
+      }
+
+      const credentialIdHash = await this.platformCryptographyTools.computePersonalHash(privateKeyBuffer, credentialId)
+
+      const typeHashes = []
+      if (isW3cCredential(credential)) {
+        for (const credentialType of credential.type) {
+          const typeHash = await this.platformCryptographyTools.computePersonalHash(privateKeyBuffer, credentialType)
+          typeHashes.push(typeHash)
+        }
+      }
+
+      const cyphertext = await this.platformCryptographyTools.encryptByPublicKey(publicKeyBuffer, credential)
+
+      encryptedCredentials.push({
+        id: credentialIdHash,
+        types: typeHashes,
+        payload: cyphertext,
+        bloomVaultIndex: credential.bloomId,
+      })
+    }
+
+    return encryptedCredentials
+  }
+
+  /**
+   * Initiate VCs migration on bloom-vault VC READ event.
+   * @param credentials
+   * @param {string} accessToken - bloom-vault access token
+   * @param {string} signature - signature of bloom-vault access token
+   */
+  async runMigration(credentials: any[], accessToken: string, signature: string): Promise<void> {
+    try {
+      const encryptedVCs = await this.encryptCredentials(credentials)
+      await this.runMigrationByChunk(encryptedVCs, 100, accessToken, signature)
+    } catch (err) {
+      console.error('Vault-migration-service initiate migration for given user call ends with error: ', err)
+    }
+  }
+
+  /**
+   * Divides given encrypted VCs list into chunks and run migration
+   * @param encryptedVCs
+   * @param chunk
+   * @param {string} accessToken - bloom-vault access token
+   * @param {string} signature - signature of bloom-vault access token
+   */
+  async runMigrationByChunk(encryptedVCs: any[], chunk: number, accessToken: string, signature: string): Promise<void> {
+    for (let i = 0, L = encryptedVCs.length; i < L; i += chunk) {
+      const chunkedVCList = encryptedVCs.slice(i, i + chunk)
+      await this.migrateCredentials(chunkedVCList, accessToken, signature)
+    }
+  }
+
+  /**
+   * Gets migration status for user with given token for specified ethereum DID
+   */
+  async getMigrationStatus(): Promise<string> {
+    const url = `migration/done/${this.bloomDid}`
+    let migrationDone: 'no' | 'yes' | 'error' = 'error'
+    try {
+      const token = await this.getAuth()
+      const response = await this.api.execute(
+        'GET',
+        url,
+        {},
+        {
+          Authorization: `Bearer ${token}`,
+        },
+      )
+      if (response) migrationDone = 'yes'
+      else migrationDone = 'no'
```

Comment:
this endpoint returns boolean in response in body, so just return result and throw an error in case of error